openmc/kinetics/solver.py

```diff
@@ -0,0 +1,831 @@
+                residual_array = openmc.kinetics.nan_inf_to_zero(residual_array)
+                num_fissile_regions = np.sum(power_old > 0.)
+                residual = np.sqrt((residual_array**2).sum() / num_fissile_regions)
+                power_old = new_power
+
+            if residual < self.outer_tolerance and iteration >= self.min_outer_iters:
+                print('  CONVERGED OUTER residual {}'.format(residual))
+                break
+            else:
+
+                print('UNCONVERGED OUTER residual {}'.format(residual))
+
+                # Increment outer iteration count
+                iteration += 1
+
+                # Run OpenMC on forward out state
+                if iteration == 1:
+                    self.run_openmc('FORWARD_OUTER')
+                else:
+                    self.run_openmc('FORWARD_OUTER', self.method)
+
+                state_fwd.extract_shape()
+
+                # Reset the inner states
+                self.copy_states('PREVIOUS_OUTER', 'FORWARD_INNER')
+                self.copy_states('PREVIOUS_OUTER', 'PREVIOUS_INNER')
+
+        # Dump data at FORWARD_OUT state to log file
+        state_fwd.dump_to_log_file
+
+        # Copy the converged forward state to the previous state
+        self.copy_states('FORWARD_OUTER', 'PREVIOUS_OUTER')
+
+        # Increment the clock time step
+        self.clock.outer_step += 1
+
+    def generate_tallies_file(self, time_point):
+
+        # Generate a new tallies file
+        tallies_file = openmc.Tallies()
+
+        # Get the MGXS library
+        mgxs_lib = self.states[time_point].mgxs_lib
+
+        # Add the tallies to the file
+        for mgxs in mgxs_lib.values():
+            tallies = mgxs.tallies.values()
+            for tally in tallies:
+                tallies_file.append(tally, True)
+
+        # Export the tallies file to xml
+        tallies_file.export_to_xml(self.directory + '/tallies.xml')
+
+    def solve(self):
+
+        # Create run directory
+        if not os.path.exists(self.directory):
+            os.makedirs(self.directory)
+
+        # Create states and run initial OpenMC on initial state
+        self.create_state('START')
+
+        # Compute the initial steady state flux
+        self.compute_initial_flux()
+
+        # Solve the transient
+        for i in range(self.clock.num_outer_steps):
+            self.take_outer_step(i)
+
+    def setup_openmc(self, time_point, method):
+
+        # Get a fresh copy of the settings file
+        settings_file = copy.deepcopy(self.settings_file)
+
+        # Create a new random seed for the xml file
+        if self.constant_seed:
+            settings_file.seed = self.seed
+        else:
+            settings_file.seed = np.random.randint(1, 1e6, 1)[0]
+
+        if self.mgxs_lib_file:
+            self.materials_file.cross_sections = './mgxs.h5'
+            self.mgxs_lib_file.export_to_hdf5(self.directory + '/mgxs.h5')
+            settings_file.energy_mode = 'multi-group'
```

Comment:
It should be the responsibility of the user to make sure their `Settings` object has this set properly, rather than having this class selectively set it